### PR TITLE
[expo-localauthentication][android] Fix missing message definition

### LINKED ISSUE #10682: [expo-local-authentication][Android] Missing message definition on type LocalAuthenticationResult
## 🐛 Bug Report

LocalAuthenticationResult doesn't provide pattern:
```javasript
{ success: false; error: string; message: string }
```

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
```javasript
"expo-local-authentication": "^9.3.0",
"react-native-unimodules": "^0.7.0",
```

### Reproducible Demo

- when I call `authenticateAsync`
```
const result = await LocalAuthentication.authenticateAsync({
    cancelLabel: 'cancel',
});
```
- if we didn't setup passcode (maybe just use `swipe`)
- result return:
```
{ message: 'KeyguardManager#isDeviceSecure() returned false';  error: 'not_enrolled'; success: false }
```
- but `LocalAuthenticationResult`'s definition doesn't define `message`:
`src/LocalAuthentication.types.ts`
```
export type LocalAuthenticationResult = { success: true } | { success: false; error: string };
```
image
 
### Steps to Reproduce
AS ABOVE

### Expected Behavior vs Actual Behavior

Expected Behavior(LocalAuthenticationResult definition):
```
export type LocalAuthenticationResult = { success: true } | { success: false; error: string } | { success: false; error: string; message: string };
```

Actual Behavior(LocalAuthenticationResult definition):
```
export type LocalAuthenticationResult = { success: true } | { success: false; error: string };
```
Do I need to create PR to resolve this simple issue?

### PR DESCRIPTION
**Pull request by mstach60161**
# Why

Fixes  #10682.

# How

- Added missing parameter warning to LocalAuthenticationResult.
- Added missing parameter warning in promise.resolve in LocalAuthenticationModule.
- Updated native component list with result warning.

# Test Plan

- Native component list.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-local-authentication/src/LocalAuthentication.types.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export type LocalAuthenticationResult = { success: true } | { success: false; error: string };
+export type LocalAuthenticationResult = { success: true } | { success: false; error: string; message: string };
```
Comment (lukmccall) [inline packages/expo-local-authentication/src/LocalAuthentication.types.ts:1]:
```suggestion
export type LocalAuthenticationResult = { success: true } | { success: false; error: string; warning?: string };
```
It should be called a warning and should be optional.

File: packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt [hunk 2]
```diff
@@ -170,6 +170,7 @@ class LocalAuthenticationModule(context: Context) : ExportedModule(context), Act
         this.promise?.resolve(Bundle().apply {
           putBoolean("success", false)
           putString("error", "app_cancel")
+          putString("message", "isAuthenitcating returned false")
```
Comment (lukmccall) [inline packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt:173]:
```suggestion
```

**Review by lukmccall — CHANGES_REQUESTED** [review]
I think it's not a good way to go. I inspected the iOS code and it sometimes returns the `warning` property, which is automatically piped to the console. We should the same here. So please rename `message` to `warning` on Android 😉

**Review by lukmccall — CHANGES_REQUESTED** [review]
You still need to change every "message" that occurs in LocationModule to "warning" in the Android codebase.

**Comment by lukmccall** [comment]
Please, resolve conflicts 😉

**Comment by lukmccall** [comment]
Please run `./gradlew :expo-local-authentication:spotlessApply` in `apps/bare-expo/android`

**Comment by mstach60161** [comment]
- [x] Please run `./gradlew :expo-local-authentication:spotlessApply` in `apps/bare-expo/android`

> BUILD SUCCESSFUL in 57s
> 3 actionable tasks: 3 executed

**Comment by lukmccall** [comment]
Bump - fix CI and resolve confilits

**Review by lukmccall — APPROVED** [review]
LGTM 🥇

**Pull request merged**